src/pages/cms/article/[path].js

```diff
@@ -0,0 +1,277 @@
+import ImageManager from "../../../components/organisms/ImageManager";
+import Button from "../../../components/atoms/Button";
+import ArticleService from "../../../services/ArticleService";
+import EditArticleMetadata from "../../../components/organisms/EditArticleMetadata";
+import Breadcrumb from "../../../components/molecules/Breadcrumb";
+import ArticleRevisionService from "../../../services/ArticleRevisionService";
+
+const Layout = styled("div")`
+	margin: 0 auto;
+	display: grid;
+	grid-template-columns: auto 275px;
+	grid-column-gap: 25px;
+`;
+
+const PublishText = styled("p")`
+	margin: 5px 0;
+	font-size: .75rem;
+`;
+
+const TextArea = styled("textarea")`
+	width: 100%;
+	height: 300px;
+	resize: vertical;
+	border: 1px solid var(--border);
+`;
+
+const Preview = styled(Article)`
+	padding: 15px;
+	border: 1px solid var(--border);
+`;
+
+const PreviewTab = {
+	CONTENT: 0,
+	PREVIEW: 1,
+	METADATA: 2
+};
+
+class ManageArticle extends Component {
+	articleService = new ArticleService();
+	revisionService = new ArticleRevisionService();
+	state = {
+		loading: true,
+		success: undefined,
+		error: undefined,
+		activeTab: PreviewTab.CONTENT,
+		activeRevision: undefined,
+		content: "",
+		articleData: {},
+		revisionsData: []
+	};
+
+	saveContent = async event => {
+		const { articleData, content } = this.state;
+
+		event.preventDefault();
+
+		try {
+			const revisionId = await this.revisionService.createArticleRevision(articleData.id, {
+				content,
+				description: articleData.revision?.description ?? "Not set",
+				tags: articleData.revision?.tags
+			});
+
+			this.setRevisionData({
+				target: {
+					value: revisionId
+				}
+			});
+
+			this.setState({
+				success: `Created and saved new revision (Revision ID: ${revisionId})`
+			});
+
+			setTimeout(() => this.setState({ success: undefined }), 3000);
+		} catch ({ message }) {
+			this.setState({
+				error: message
+			});
+		}
+	}
+
+	setContent = event => {
+		this.setState({
+			content: event.target.value
+		});
+	}
+
+	setRevisionData = event => {
+		const revisionId = +event.target.value;
+		const { revisionsData } = this.state;
+
+		const activeRevision = revisionsData.find(revision => revision.id === revisionId);
+
+		if (!activeRevision) {
+			return this.getData(revisionId);
+		}
+
+		this.setState({
+			activeRevision,
+			content: activeRevision.content
+		});
+	}
+
+	setActiveTab = tab => {
+		this.setState({
+			activeTab: tab
+		});
+	}
+
+	getData = async revisionId => {
+		const articleId = +window.location.pathname.replace("/cms/article/", "");
+
+		try {
+			this.setState({
+				loading: true,
+				error: undefined
+			});
+
+			const articleData = await this.articleService.getArticleById(articleId);
+			const revisionsData = await this.articleService.getArticleRevisions(articleId);
+			const activeRevision = revisionsData.find(revision => revision.id === revisionId) ?? articleData.revision;
```

Comment:
I personally think it's best to pick the latest revision by default (even if not active) since this would be the most natural flow. Think about if you're part way through an update to a revision and keep coming back - if the default option chosen is the "live" article, you may inadvertently edit the wrong version. Happy to hear your thoughts though!